File: Ejercicios_POO3.1_Java/src/Ejercicio3_JuegoRol/Guerrero.java
package Ejercicio3_JuegoRol;

public class Guerrero extends Personaje {
    private final int poderAtaque = 10;

    public Guerrero(String nombre, int puntosVida) {
        super(nombre, puntosVida);
    }

    @Override
    public int atacar() {
        System.out.println(nombre + " ataca con un golpe poderoso!");
        return poderAtaque;
    }
}

File: Ejercicios_POO3.1_Kotlin/src/Ejercicio3_JuegoRol/Guerrero.kt
package Ejercicio3_JuegoRol

class Guerrero(nombre: String, puntosVida: Int) : Personaje(nombre, puntosVida) {
    private val poderAtaque = 10

    override fun atacar(): Int {
        println("$nombre ataca con un golpe poderoso!")
        return poderAtaque
    }
}